haystack/components/generators/chat/azure.py

```diff
@@ -32,30 +32,33 @@ class AzureOpenAIChatGenerator(OpenAIChatGenerator):
-                                azure_deployment="<this a model name, e.g. gpt-35-turbo>")
+    client = AzureOpenAIGenerator(
+        azure_endpoint="<Your Azure endpoint e.g. `https://your-company.azure.openai.com/>",
+        api_key=Secret.from_token("<your-api-key>"),
+        azure_deployment="<this a model name, e.g. gpt-35-turbo>")
     response = client.run(messages)
     print(response)
+    ```
 
-    >>{'replies': [ChatMessage(content='Natural Language Processing (NLP) is a branch of artificial intelligence
-    >>that focuses on enabling computers to understand, interpret, and generate human language in a way that is
-    >>meaningful and useful.', role=<ChatRole.ASSISTANT: 'assistant'>, name=None,
-    >>meta={'model': 'gpt-3.5-turbo-0613', 'index': 0, 'finish_reason': 'stop',
-    >>'usage': {'prompt_tokens': 15, 'completion_tokens': 36, 'total_tokens': 51}})]}
-
+    ```
+    {'replies':
+        [ChatMessage(content='Natural Language Processing (NLP) is a branch of artificial intelligence that focuses on
+         enabling computers to understand, interpret, and generate human language in a way that is meaningful and useful.',
+         role=<ChatRole.ASSISTANT: 'assistant'>, name=None,
+         meta={'model': 'gpt-3.5-turbo-0613', 'index': 0, 'finish_reason': 'stop',
+         'usage': {'prompt_tokens': 15, 'completion_tokens': 36, 'total_tokens': 51}})]
+    }
     ```
 
      Key Features and Compatibility:
-         - **Primary Compatibility**: Designed to work seamlessly with the OpenAI API Chat Completion endpoint
-            and gpt-4 and gpt-3.5-turbo family of models.
-         - **Streaming Support**: Supports streaming responses from the OpenAI API Chat Completion endpoint.
-         - **Customizability**: Supports all parameters supported by the OpenAI API Chat Completion endpoint.
+       - Primary Compatibility: Designed to work seamlessly with the OpenAI API Chat Completion endpoint and gpt-4 and gpt-3.5-turbo family of models.
+       - Streaming Support: Supports streaming responses from the OpenAI API Chat Completion endpoint.
+       - Customizability: Supports all parameters supported by the OpenAI API Chat Completion endpoint.
 
      Input and Output Format:
-         - **ChatMessage Format**: This component uses the ChatMessage format for structuring both input and output,
-           ensuring coherent and contextually relevant responses in chat-based text generation scenarios. Details on the
-           ChatMessage format can be found at: https://github.com/openai/openai-python/blob/main/chatml.md.
+       - ChatMessage Format: This component uses the ChatMessage format for structuring both input and output, ensuring
+         coherent and contextually relevant responses in chat-based text generation scenarios.
+     - Details on the ChatMessage format can be found [here](https://github.com/openai/openai-python/blob/main/chatml.md.)
```

Comment:
This URL is outdated (was previously deprecated by OpenAI).
I would refer to our docs: https://docs.haystack.deepset.ai/v2.0/docs/data-classes#chatmessage